### ISSUE TITLE
[REVIEW]: DOENOVA: Simple DOE and ANOVA package for Python 3

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1685]
**Submitting author:** @fblavoie (<a href="http://orcid.org/0000-0003-4574-1315">Francis B Lavoie</a>)
**Repository:** <a href="https://github.com/fblavoie/doenova" target ="_blank">https://github.com/fblavoie/doenova</a>
**Version:** 0.1.3
**Editor:** @arfon
**Reviewers:** @klmedeiros, @usethedata
**Archive:** Pending

## Status

[![status](https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f/status.svg)](https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f"><img src="https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f/status.svg)](https://joss.theoj.org/papers/a9f075ecbce040d20cba04207d1a537f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/fblavoie/doenova" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@klmedeiros, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @usethedata know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/fblavoie/doenova">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fblavoie) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/fblavoie/doenova">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@fblavoie) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by klmedeiros** [comment]
i'm close to done on this, but i have a few issues blocking acceptance, posted here: https://github.com/fblavoie/doenova/issues/2

also, is it an issue that some of what is listed in the review lives on the accompanying website, not in the repo?

**Comment by fblavoie** [comment]
@klmedeiros ,
Contributing instructions and test scripts were indeed missing. They were added. I have also uploaded the website on the github repo.

**Comment by klmedeiros** [comment]
@usethedata This review is completed and the paper is ready to be accepted.

**Comment by usethedata** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by usethedata** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01685/joss.01685/10.21105.joss.01685.pdf)

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.2307/2332195 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1109/MCSE.2007.58 is INVALID because of 'https://doi.org/' prefix
```

**Comment by usethedata** [comment]
@fblavoie -- in your references.bib file, you need just the doi, not the url prefix.  For example, with the plackett1946design reference, you have `doi={https://doi.org/10.2307/2332195}`  This needs to be `doi={10.2307/2332195}`.  Can you fix this and the oliphant2007python reference?

**Comment by usethedata** [comment]
@whedon add @usethedata as reviewer

**Comment by whedon** [comment]
OK, @usethedata is now a reviewer

**Comment by danielskatz** [comment]
👋 @usethedata - how is this going?

**Comment by usethedata** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01685/joss.01685/10.21105.joss.01685.pdf)

**Comment by usethedata** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- https://doi.org/10.2307/2332195 is OK
- https://doi.org/10.1109/MCSE.2007.58 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by usethedata** [comment]
My review is complete.  Nice work, and well-written.  This is a tool I expect to make use of from time to time.

**Comment by usethedata** [comment]
@fblavoie I gave you incomplete/inaccurate information on fixing the DOI in the bibliography.  I've submitted a pull request to your repo that should fix the issue.  I also suggest closing the issue you have open from klmedeiros review. Once those are done, then I believe the paper is ready for me to request acceptance.

My apologies for the delays on this end.  I've had some family needs that have taken a lot of time.

**Comment by usethedata** [comment]
@fblavoie checking in -- did you have a chance to look at the the pull request I submitted on your repo to correct the bibliography?  Thanks.

**Comment by labarba** [comment]
@usethedata @fblavoie — it looks like this submission needs a final push. Can you let us know the estimated time to get to the finish line? 🚀

**Comment by arfon** [comment]
👋 all, I'm going to step in here and try and wrap up this submission on behalf of @usethedata.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01685/joss.01685/10.21105.joss.01685.pdf)

**Comment by arfon** [comment]
@fblavoie - please merge this pull request that makes a number of small changes to your paper: https://github.com/fblavoie/doenova/pull/4

When you have done this could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by arfon** [comment]
For any EiCs on rotation, I emailed @fblavoie on the 1st February with this request also.

**Comment by danielskatz** [comment]
👋 @arfon - should we change the editor to be you to better track this and finish it?

**Comment by arfon** [comment]
@whedon assign @arfon as editor

**Comment by arfon** [comment]
It's now more than four months since [we last heard](https://github.com/openjournals/joss-reviews/issues/1685#issuecomment-537184269) from the author (@fblavoie). If there's no response by March 1st (which will be 5 months since last contact) then I will close this issue and withdraw the paper on their behalf, assuming that they have lost interest in pursuing this publication with JOSS.

**Comment by arfon** [comment]
@klmedeiros, @usethedata - thanks again for your reviews here. Unfortunately as the author has been absent now for 5 months I'm think the only reasonable conclusion I can make is that the author is no-longer interested in pursuing this paper in JOSS. As such, I'm going to go ahead and withdraw the submission on their behalf.

**Comment by arfon** [comment]
@whedon withdraw

**Comment by whedon** [comment]
Paper withdrawn.
